plugins/security/nexus-ldap-plugin/src/main/java/org/sonatype/security/realms/ldap/internal/persist/LdapConfigurationManager.java

```diff
@@ -14,28 +14,48 @@
 
 import java.util.List;
 
-import com.sonatype.security.ldap.realms.persist.model.CLdapServerConfiguration;
-
-import org.sonatype.configuration.validation.InvalidConfigurationException;
+import org.sonatype.security.realms.ldap.internal.persist.entity.LdapConfiguration;
 
+/**
+ * Component responsible for handling caching, eventing and driving persistence using {@link LdapConfigurationSource}.
+ */
 public interface LdapConfigurationManager
 {
-
+  /**
+   * Clears in-memory cache of LDAP server configuration.
+   */
   void clearCache();
 
-  List<CLdapServerConfiguration> listLdapServerConfigurations();
-
-  CLdapServerConfiguration getLdapServerConfiguration(String id)
-      throws InvalidConfigurationException, LdapServerNotFoundException;
-
-  void updateLdapServerConfiguration(CLdapServerConfiguration ldapServerConfiguration)
-      throws InvalidConfigurationException, LdapServerNotFoundException;
-
-  void addLdapServerConfiguration(CLdapServerConfiguration ldapServerConfiguration)
-      throws InvalidConfigurationException;
-
-  void deleteLdapServerConfiguration(String id) throws InvalidConfigurationException, LdapServerNotFoundException;
-
-  void setServerOrder(List<String> orderdServerIds) throws InvalidConfigurationException;
-
+  /**
+   * Lists defined LDAP server configurations.
+   */
+  List<LdapConfiguration> listLdapServerConfigurations();
+
+  /**
+   * Returns LDAP server configuration by it's ID, never {@code null}. If not found, exception will be thrown.
+   */
+  LdapConfiguration getLdapServerConfiguration(String id) throws LdapServerNotFoundException;
```

Comment:
This iface exists from before the change, only javadoc was added. Did not want to refactor all the existing bits too, as we would be lost in diffs (not related to the orient change). Still, if really eager for this change, could be done.